### PR TITLE
Accessibility issues for facility locator manual geolocation

### LINKED ISSUE #21976: Address usability/accessibility for  "Use my location"
@mmiddaugh commented on [Tue Mar 23 2021](https://github.com/department-of-veterans-affairs/va.gov-vfs-teams/issues/55)

@mmiddaugh commented on [Tue Mar 09 2021](https://github.com/department-of-veterans-affairs/va.gov-team/issues/21247)

## Issue Description
As a keyboard user, I want to be able to use the space bar to activate "Use my location" (rather than just the enter key) so that it is consistent with the usual keyboard navigation. 
As a Veteran using voice assistive technology, I want to be able to ask my tech to "Click on the use my location button" instead of "Click on the use my location link." so that I can access it better.  

"Use my location" should be written as a button <button> instead of a link <a> since the user is not navigating to a new page

---
## Tasks
- [x] Write "Use my location" as a button. 
- [x] Fix visual behavior of clear input and use my location button

## Acceptance Criteria (this includes feedback given in #17319)
- [x] Veterans navigating the Facility Locator using a keyboard can activate "Use my location" with the space bar. 
- [x] Veterans using voice assistive technology ask the tech to "Click on the Use my location button" instead of "Click on the Use my location link."
- [x] Clear input and use my location buttons do not shift to the right
- [x] The screen reader reads "finding your location" text while geolocation is in progress. 
- [x] "Use my location" should receive focus and be announced before "City state or postal code".
- [x] When geolocation completes, the focus goes to the City, state, postal code field (whether or not approval was given in the browser modal) and screenreader announces the field
- [x] If the Veteran moves into another field during the geolocation search process, that field receives focus (it is not "stolen" to focus on the city, state field)
- [x] If geolocation fails for any reason [an error message appears in a modal](https://design.va.gov/storybook/?path=/docs/components-modal--default): "Something went wrong. Please enter your location."
- [x] When the modal appears, the tab order is locked, the error message modal receives focus and is announced. 
- [x] Modal is dismissable by clicking the x and focus returns to the "City state postal field"
- [x] Data layer in place to track "clicks" on the feature 
- [x] Data layer to track the number of permission-related errors vs other types (needs analytics input)

https://dsva.slack.com/archives/C0FQSS30V/p1615814186014400?thread_ts=1615591295.004600&cid=C0FQSS30V

https://recordit.co/BmJjt9fWsS

---
## How to configure this issue
- [ ] **Attached to a Milestone** (when will this be completed?)
- [ ] **Attached to an Epic** (what body of work is this a part of?)
- [ ] **Labeled with Team** (`product support`, `analytics-insights`, `operations`, `service-design`, `tools-be`, `tools-fe`)
- [ ] **Labeled with Practice Area** (`backend`, `frontend`, `devops`, `design`, `research`, `product`, `ia`, `qa`, `analytics`, `contact center`, `research`, `accessibility`, `content`)
- [ ] **Labeled with Type** (`bug`, `request`, `discovery`, `documentation`, etc.)

---

@tlei123 commented on [Fri Mar 12 2021](https://github.com/department-of-veterans-affairs/va.gov-team/issues/21247#issuecomment-797778820)

hi @lildziner , would you expect to have updated designs for this button?

if yes, will you be updating these two places in Zeplin?
- [Mobile](https://app.zeplin.io/project/600877fdb269d8a4f1f44efd/screen/600879364afaa6a6a2a8a827)
- [Desktop](https://app.zeplin.io/project/600877fdb269d8a4f1f44efd/screen/6008792b280d84a85bc6c015)

---

@mmiddaugh commented on [Tue Mar 23 2021](https://github.com/department-of-veterans-affairs/va.gov-team/issues/21247#issuecomment-805246161)

We're not updating the design @tlei123 - just the coding

### PR DESCRIPTION
**Pull request by mpelzsherman**
Resolves https://github.com/department-of-veterans-affairs/va.gov-team/issues/21976

### CODE REVIEW

File: src/applications/facility-locator/components/SearchControls.jsx [hunk 1]
```diff
@@ -176,10 +190,11 @@ class SearchControls extends Component {
         <div className="input-container">
           <input
             id="street-city-state-zip"
+            ref={locationInputFieldRef}
```
Comment (zacharymorel) [inline src/applications/facility-locator/components/SearchControls.jsx:193]: So that's how we use useRef!

**Review by zacharymorel — APPROVED** [review]
This looks great! Nice refactoring.

File: src/applications/facility-locator/actions/index.js [hunk 2]
```diff
@@ -501,6 +501,7 @@ export const getProviderSpecialties = () => async dispatch => {
 };
 
 export const geolocateUser = () => async dispatch => {
+  const GEOLOCATION_TIMEOUT = 10000;
```
Comment (brangi) [inline src/applications/facility-locator/actions/index.js:504]: can we add this in our constants file ?

Comment (mpelzsherman) [reply src/applications/facility-locator/actions/index.js:504]: I think it's OK to leave it here since it's only used here.

**Review by brangi — APPROVED** [review]

**Pull request merged**